assembl/views/discussion/views.py

```diff
@@ -374,6 +394,54 @@ def purl_post(request):
     return home_view(request)
 
 
+@view_config(route_name='join', request_method='GET', renderer='assembl:templates/join.jinja2')
+def join_discussion(request):
+    user_id = authenticated_userid(request)
+    if not user_id:
+        return HTTPSeeOther(request.route_url(
+            'contextual_react_login', **request.matchdict))
+    discussion = discussion_from_request(request)
+    if not discussion:
+        raise HTTPNotFound("No such discussion")
+    user = User.get(user_id)
+    if not user:
+        raise HTTPClientError("No such user")
+    roles = get_roles(user_id, discussion.id)
+    if roles:
+        # already joined.
+        # bare_slug will reroute to v1 or v2
+        return HTTPSeeOther(request.route_url(
+            'bare_slug', **request.matchdict))
+    # ok, show it.
+    ut = discussion.get_participant_template()[0]
+    localizer = request.localizer
+    notifications = [localizer.translate(n.get_human_readable_description())
+                     for n in ut.notification_subscriptions
+                     if n.status == NotificationSubscriptionStatus.ACTIVE]
+    data = get_default_context(request)
+    data['notifications'] = notifications
+    return data
+
+
+@view_config(route_name='join', request_method='POST')
+def do_join_discussion(request):
```

Comment:
As the ultimate response of this view is to redirect to either a login page or throw an error, we should utilize the new error page that was created for exactly these cases. Instead of returning `HTTPUnauthorized` or others, should redirect to error page, and pass the error in.

Downstream, that view can respond to the appropriate `Error` with a client-friendly view (this is not done for all cases, only 404, 401 and 500 I recall)

Reply:
You have an error page now? Then I think that the proper thing to do is to have HTTPUnauthorized go to that error page! Pyramid allows that. Let's discuss it.

Reply:
Yup, I discussed it with you a long while back, when logged in users wanted to access a private discussion. Yes, let's discuss when you have a moment.

Reply:
ok now that we have a generic error page, this code stands.